irohad/ametsuchi/impl/postgres_query_executor.cpp

```diff
@@ -982,67 +995,193 @@ namespace iroha {
+                          select data
+                          from account
+                          where account_id = :account_id
+                      )) data_by_writer,
+                  jsonb_each(data_by_writer.value) plain_data
+                  where
+                      coalesce(data_by_writer.key = :writer, true) and
+                      coalesce(plain_data.key = :key, true)
+                  order by data_by_writer.key asc, plain_data.key asc
+              ) t
+          ),
+          page_limits as (
+              select start.rn as start, start.rn + :page_size as end
+                  from (
+                      select rn
+                      from filtered_plain_data
+                      where
+                          coalesce(writer = :first_record_writer, true) and
+                          coalesce(key = :first_record_key, true)
+                      limit 1
+                  ) start
+          ),
+          total_number as (select count(1) total_number from filtered_plain_data),
+          next_record as (
+              select writer, key
+              from
+                  filtered_plain_data,
+                  page_limits
+              where rn = page_limits.end
+          ),
+          page as (
+              select json_object_agg(writer, data_by_writer) json
+              from (
+                  select writer, json_object_agg(key, value) data_by_writer
+                  from
+                      filtered_plain_data,
+                      page_limits
+                  where
+                      rn >= page_limits.start and
+                      coalesce(rn < page_limits.end, true)
+                  group by writer
+              ) t
+          ),
+          target_account_exists as (
+            select count(1) val
+            from account
+            where account_id = :account_id
+          )
+          select
+              page.json json,
+              total_number,
+              next_record.writer next_writer,
+              next_record.key next_key,
+              target_account_exists.val target_account_exists
+          from
+              page
+              left join total_number on true
+              left join next_record on true
+              right join target_account_exists on true
+      )
+      select detail.*, perm from detail
+      right join has_perms on true
       )")
                   % hasQueryPermission(creator_id_,
                                        q.accountId(),
                                        Role::kGetMyAccDetail,
                                        Role::kGetAllAccDetail,
-                                       Role::kGetDomainAccDetail)
-                  % query_detail)
+                                       Role::kGetDomainAccDetail))
                      .str();
 
+      const auto writer = q.writer();
+      const auto key = q.key();
+      boost::optional<std::string> first_record_writer;
```

Comment:
```suggestion
      boost::optional<shared_model::interface::types::AccountDetailRecordId> first_record_id;
```
and remove `first_record_key`

Reply:
no, these are for the use in sql query